src/v/cluster/topic_recovery_validator.cc

```diff
@@ -0,0 +1,307 @@
+            co_return ss::coroutine::exception(ex);
+        }
+        op_logger_.error(
+          "manifest {} metadata check threw exception: {}, "
+          "validation not ok",
+          get_path(),
+          ex);
+        co_return validation_result::anomaly_detected;
+    }
+
+    // examine anomalies, some are real failures while other can be
+    // tolerated
+    auto anomalies = anomalies_fut.get();
+    if (anomalies.status == cloud_storage::scrub_status::failed) {
+        // download issue. could be a configuration issue or a s3 issue
+        op_logger_.error(
+          "manifest {} metadata check download failure, validation not ok",
+          get_path());
+        co_return validation_result::download_issue;
+    }
+
+    if (!anomalies.detected.has_value()) {
+        // no anomalies detected, everything is ok up to segment depth
+        op_logger_.info("manifest metadata check passed, validation ok");
+        co_return validation_result::passed;
+    }
+
+    // this unpack is to trigger a compilation error if a new field is added
+    // to cloud_storage::anomalies. If a new anomalies is added, it should
+    // be also handled here
+    auto const& [no_partition_manifest, missing_spillovers, missing_segments, segment_anomalies, ignore1, ignore2, ignore3, ignore4]
+      = anomalies.detected;
+
+    if (no_partition_manifest) {
+        // missing manifest anomaly can be ok: this means the partition did
+        // not have time to upload data
+        op_logger_.info("manifest metadata check: no manifest, validation ok");
+        co_return validation_result::missing_manifest;
+    }
+
+    if (!missing_segments.empty()) {
+        // missing segments could be problematic and need an operator fixing
+        // the manifest before proceeding
+        op_logger_.error(
+          "manifest metadata check: missing segment, validation not ok");
+        co_return validation_result::anomaly_detected;
+    }
+
+    if (!missing_spillovers.empty()) {
+        // missing spillover manifest are not as problematic as missing
+        // segments, but still require manual intervention to fix the
+        // manifest for a correct operation
+        op_logger_.error("manifest metadata check: missing spillover "
+                         "manifests, validation not ok");
+        co_return validation_result::anomaly_detected;
+    }
+
+    // classify anomaly_meta as failure or ok (not problematic for the
+    // recovery use case) true: failure, false: passed
+    constexpr static auto is_fatal_anomaly =
+      [](cloud_storage::anomaly_meta const& am) {
+          using enum cloud_storage::anomaly_type;
+          switch (am.type) {
+          case missing_delta:
+          case non_monotonical_delta:
+          case end_delta_smaller:
+          case committed_smaller:
+              return true;
+          case offset_gap:
+          case offset_overlap:
+              return false;
```

Comment:
At least from the metrics, there's an existing categorization of "high" and "low' severity, but it doesn't quite seem to match https://github.com/redpanda-data/redpanda/blob/24e7b780da6b0ac31174935def6e175d004652b2/src/v/cluster/partition_probe.cc#L368-L443. Is there a different categorization for what constitutes fatal?

Reply:
changed to match